lib/creation_notifier.rb

```diff
@@ -38,9 +38,13 @@ def notify_recipients
                                    recipient_emails_off: false).pluck(:user_id)
     recip_ids.each do |userid|
       if self.collections.empty? || self.collections.first.nil?
-        UserMailer.recipient_notification(userid, self.id).deliver_after_commit
+        I18n.with_locale(User.find(userid).preference.locale.iso) do
```

Comment:
Instead of finding the preference from the userid again, we can go about this differently. If you remove the `pluck` from the `recip_ids` line so you get the full preference records then you can loop through the preferences and get both the user as well as the locale directly from the preference